Add VIP wBTC vault

timongll opened the pull request:
-adds rVIP-wBTC vault
-basic template for future VIP vaults
<img width="1725" alt="image" src="https://user-images.githubusercontent.com/25440724/210965600-a8c720c0-1874-467d-b9a5-d27ff1ae6233.png">

Sahilgat commented on shared/src/hooks/useFetchEarnVaultData.ts:
```diff
@@ -131,6 +136,22 @@ const useFetchEarnVaultData = (): V2VaultData => {
+          : Promise.resolve(BigNumber.from(0));
+
+        let roundPricePerShare: BigNumber = BigNumber.from(0);
+        await roundPricePerSharePromise.then((v) => (roundPricePerShare = v));
```
Could we write this as something like `const roundPricePerShare = vaultState.round ? await contract.roundPricePerShare(vaultState.round - 1) : BigNumber.from(0)`?

timongll commented on treasury/src/pages/DepositPage/DepositPage.tsx:
```diff
@@ -190,6 +193,20 @@ const DepositPage = () => {
+              2
+            )
+          ),
+          parseFloat(formatSignificantDecimals(formatUnits(cap, decimals))),
```
I've added a refactor, will refactor other parts of the repo in future PR

timongll commented:
addressed comments in [26dd9cc](https://github.com/ribbon-finance/ribbon-frontend/pull/653/commits/26dd9cc72ccef48c7ee0be30758e531cf067cdce) and [1544801](https://github.com/ribbon-finance/ribbon-frontend/pull/653/commits/15448013a8e353d0ed8f976c3041f09faa33a61b)
